### ISSUE TITLE
Dotnet build fails for Windows arm64

### ISSUE THREAD

**Issue by kunalspathak** [dotnet/roslyn#42393]
**Version Used**: 
Microsoft (R) Visual C# Compiler version 3.6.0-2.20153.4 (6f2112b5)

** Platform **
Windows ARM64 netcoreapp5.0

**Steps to Reproduce**:

1.  dotnet new console
2. dotnet build created_project.csproj

**Expected Behavior**:
Project builds

**Actual Behavior**:
```
CSC : error CS0041: Unexpected error writing debug information -- 'An attempt was made to load a program with an incorrect format. (0x8007000B)
```

The reason being that CSC tries to load amd64 version of `Microsoft.DiaSymReader.Native.amd64.dll ` to write symbols.

@jaredpar  suggested to use `/debug:pdbonly` which makes it works, but I am trying to run BenchmarkDotNet which builds project on fly and it doesn't pass that flag to individual benchmark project causing it to fail with the above error.

**Comment by kunalspathak** [comment]
//cc @tmat

**Comment by tmat** [comment]
Blocked on https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1034547

**Comment by adamsitnik** [comment]
@tmat The blocking issue has been solved a year ago. What would it take to fix it in Roslyn now?

**Comment by adamsitnik** [comment]
Update: I've found the places in the code that would need to be updated:

https://github.com/dotnet/roslyn/blob/main/eng/targets/DiaSymReaderNative.targets
https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/DiaSymReader/SymUnmanagedFactory.cs
https://github.com/dotnet/roslyn/blob/c8ebc8682889b395fcb84c85bf4ff54577377d26/src/NuGet/Microsoft.Net.Compilers.Toolset/DesktopCompilerArtifacts.targets#L32-L33

But it seems that there are no arm CI legs to verify such changes as of now: https://github.com/dotnet/roslyn/issues/59421#issuecomment-1042474754

**Comment by tmat** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/59421
